src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Types.cs

```diff
@@ -240,7 +240,9 @@ private void AddNameAndTypeArgumentsOrParameters(INamedTypeSymbol symbol)
             }
             else if (symbol.IsTupleType)
             {
-                if (CanUseTupleTypeName(symbol))
+                // If top level tuple uses non-default names, there is no way to preserve them
+                // unless we use tuple syntax for the type. So, we give them priority.
+                if (HasNonDefaultTupleElementNames(symbol) || CanUseTupleTypeName(symbol))
```

Comment:
That would violate the language grammar as it is defined today. Besides, the case when both the top level and the nested levels have friendly names cannot originate from source. So, I think it is fine to drop some information in this case.  

---
In reply to: [62353540](https://github.com/dotnet/roslyn/pull/11136#discussion_r62353540) [](ancestors = 62353540)